src/main/java/com/odysseusinc/arachne/datanode/service/messaging/PathwayRequestHandler.java

```diff
@@ -0,0 +1,179 @@
+
+    		List<Atlas> atlases27 = AtlasUtils.filterAtlasByVersion27(atlasList);
+        List<Pathway> pathways = atlasService.<AtlasClient2_7, Pathway>execute(atlases27, c -> c.getPathways(PAGE_SIZE).getContent());
+        return pathways.stream()
+                .map(pathway -> conversionService.convert(pathway, CommonPathwayDTO.class))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<MultipartFile> getAtlasObject(String guid) {
+
+        return commonEntityService.findByGuid(guid).map(entity -> {
+            Atlas origin = entity.getOrigin();
+            JsonNode design = atlasService.<AtlasClient2_7, JsonNode>execute(origin, atlasClient -> atlasClient.exportPathwayDesign(entity.getLocalId()));
+            String analysisName = design.get("name").asText();
+            String packageName = String.format("CohortPathways%d", entity.getLocalId());
+            List<MultipartFile> files = new ArrayList<>();
+            try {
+			 	byte[] packageContent = atlasService.hydrateAnalysis(design, packageName, SKELETON_RESOURCE);
+			 	String packageFileName = String.format("CohortPathways - %s.zip", analysisName);
+			 	MultipartFile packageFile = new MockMultipartFile(packageFileName, packageFileName, MediaType.APPLICATION_OCTET_STREAM_VALUE,
+						packageContent);
+				// Target cohorts
+				List<CohortDefinition> cohortDefinitions = new ArrayList<>();
+				JsonNode targetCohortsNode = design.get("targetCohorts");
+				if (targetCohortsNode instanceof ArrayNode) {
+					cohortDefinitions.addAll(addCohorts(origin, files, (ArrayNode) targetCohortsNode));
+				}
+				// Event cohorts
+				JsonNode eventCohortsNode = design.get("eventCohorts");
+				if (eventCohortsNode instanceof ArrayNode) {
+					cohortDefinitions.addAll(addCohorts(origin, files, (ArrayNode) eventCohortsNode));
+				}
+//				files.add(designFile);
+				int localId = entity.getLocalId();
+				files.add(packageFile);
+				files.add(getRunner(cohortDefinitions, localId, packageName, String.format("pathwaysAnalysis_%d", localId),
+						packageFileName));
+				return files.stream().filter(Objects::nonNull).collect(Collectors.toList());
+			} catch (IOException e) {
+            	LOGGER.error(PATHWAY_BUILD_ERROR, e);
+            	throw new RuntimeIOException(PATHWAY_BUILD_ERROR, e);
+            }
+        }).orElse(null);
+    }
+
+    @Override
+    public CommonAnalysisType getAnalysisType() {
+
+        return CommonAnalysisType.COHORT_PATHWAY;
+    }
+
+    @Override
+    public void sendResponse(List<MultipartFile> response, String id) {
+
+        centralClient.sendCommonEntityResponse(id, response.toArray(new MultipartFile[0]));
+    }
+
+    private List<CohortDefinition> addCohorts(Atlas origin, List<MultipartFile> files, ArrayNode cohorts) {
+
+    	List<CohortDefinition> cohortDefinitions = new ArrayList<>();
+			cohorts.forEach(cohort -> {
+				int id = cohort.get("id").intValue();
```

Comment:
tbh, I guess this class do to much, I would pull out the conversation(from JsonNode -> Cohort, jsonNode-> Design ) in separate conversation class.

Reply:
It's discussable
In one hand this really could be separated
but in another do we need to create separate class which is used only from one place?